pipeline-yaml_5b5744d40428631d7a8940a9.md

```diff
@@ -371,10 +372,59 @@ Error: "There are at least two blocks with same name: Build Go project"
 Semaphore assigns its own unique names to nameless `blocks` items, which are
 displayed in the Semaphore 2.0 UI.
 
+### dependencies in blocks
+
+The `dependencies` property of a block is used to define the flow of execution
+between blocks within a pipeline.
+
+To specify that block with name `Block C` needs to be run once the
+blocks with names `Block A` and `Block B` finish, the `dependencies` property of
+the block with name `Block C` needs to be populated with corresponding names
+of the blocks which block with name `Block C` depends on.
+In the following example blocks with names `Block A` and `Block B` would be executed in parallel at the
+very beginning of a pipeline.
+
+``` yaml
+version: "v1.0"
+name: Pipeline with dependencies
+agent:
+  machine:
+    type: e1-standard-2
+    os_image: ubuntu1804
+
+blocks:
+  - name: "Block A"
+    dependencies: []
+    task:
+      jobs:
+      - name: "Job A"
+        commands:
+          - echo "output"
+  - name: "Block B"
+    dependencies: []
+    task:
+      jobs:
+      - name: "Job B"
+        commands:
+          - echo "output"
+  - name: "Block C"
+    dependencies: ["Block A", "Block B"]
+    task:
+      jobs:
+      - name: "Job C"
+        commands:
+          - echo "output"
+```
+
+*Note*: When `dependencies` property is used to define the names of the blocks which certain block
+depends on, property becomes mandatory for all other blocks within a pipeline
+definition. Otherwise the YAML validation error is being thrown.
```

Comment:
This is hard to understand. Can you provide an example of what wouldn't work?

Reply:
I think it means that if you write `dependencies` in one block, then you have to write it in each block?

Reply:
Yes. I reduced the number of words and added one example when the specification is invalid.